integration-tests/src/test/java/com/datastax/oss/driver/core/resolver/MockResolverIT.java

```diff
@@ -59,47 +58,148 @@ public class MockResolverIT {
+              .withBoolean(TypedDriverOption.RESOLVE_CONTACT_POINTS.getRawOption(), false)
+              .withBoolean(TypedDriverOption.RECONNECT_ON_INIT.getRawOption(), true)
+              .withStringList(
+                  TypedDriverOption.CONTACT_POINTS.getRawOption(),
+                  Collections.singletonList("node-1.cluster.fake:9042"))
+              .build();
+
+      CqlSessionBuilder builder = new CqlSessionBuilder().withConfigLoader(loader);
+      try (CqlSession session = builder.build()) {
+        ResultSet rs = session.execute("SELECT * FROM system.local");
+        List<Row> rows = rs.all();
+        assertThat(rows).hasSize(1);
+        LOG.trace("system.local contents: {}", rows.get(0).getFormattedContents());
+        Collection<Node> nodes = session.getMetadata().getNodes().values();
+        for (Node node : nodes) {
+          LOG.trace("Found metadata node: {}", node);
+        }
+        Set<Node> filteredNodes;
+        filteredNodes =
+            nodes.stream()
+                .filter(x -> x.toString().contains("node-1.cluster.fake"))
+                .collect(Collectors.toSet());
+        assertThat(filteredNodes).hasSize(1);
+        InetSocketAddress address =
+            (InetSocketAddress) filteredNodes.iterator().next().getEndPoint().resolve();
+        assertTrue(address.isUnresolved());
+      }
+    }
+  }
 
-    MockResolverFactory resolverFactory = new MockResolverFactory();
-    resolverFactory.updateResponse(
-        "node-1.cluster.fake",
-        new ValidResponse(new InetAddress[] {getNodeInetAddress(ccmBridge, 1)}));
-    ResolverProvider.setDefaultResolverFactory(resolverFactory);
+  public void replace_cluster_test() {
 
     DriverConfigLoader loader =
         new DefaultProgrammaticDriverConfigLoaderBuilder()
             .withBoolean(TypedDriverOption.RESOLVE_CONTACT_POINTS.getRawOption(), false)
             .withBoolean(TypedDriverOption.RECONNECT_ON_INIT.getRawOption(), true)
             .withStringList(
                 TypedDriverOption.CONTACT_POINTS.getRawOption(),
-                Collections.singletonList("node-1.cluster.fake:9042"))
+                Collections.singletonList("test.cluster.fake:9042"))
             .build();
 
     CqlSessionBuilder builder = new CqlSessionBuilder().withConfigLoader(loader);
-    try (CqlSession session = builder.build()) {
+    CqlSession session;
+
+    try (CcmBridge ccmBridge = CcmBridge.builder().withNodes(3).withIpPrefix("127.0.1.").build()) {
+      MockResolverFactory mockResolverFactory = new MockResolverFactory();
+      mockResolverFactory.updateResponse(
+          "test.cluster.fake",
+          new ValidResponse(
+              new InetAddress[] {
+                getNodeInetAddress(ccmBridge, 1),
+                getNodeInetAddress(ccmBridge, 2),
+                getNodeInetAddress(ccmBridge, 3)
+              }));
+      ResolverProvider.setDefaultResolverFactory(mockResolverFactory);
```

Comment:
We need to make sure that there is only one instance of `MockResolverFactory` that is created and registered.
In the same way we do that for `CCM_RULE`.
I would recomment to have some method `CreateAndRegisterMockResolverFactory` that does that.

Also we need to move `MockResolverFactory` initialization to the class level.

Reply:
I'm not sure If I understand the intent here. Don't we want to have different test methods in this class that test different scenarios? So in that case we should be able to set different resolvers for each method.

And for the scenario where we kill old cluster and start new with different IPs don't we want to have the ability to change the resolver to the new IPs too? I think this change would prevent that.

Reply:
> I'm not sure If I understand the intent here. Don't we want to have different test methods in this class that test different scenarios? So in that case we should be able to set different resolvers for each test method.

Yes we won't.
Not we don't need different resolvers, test can work with resolver ammedning dns records that are relevant only to the test. 

> 
> And for the scenario where we kill old cluster and start new with different IPs don't we want to have the ability to change the resolver to the new IPs too? I think this change would prevent that.

how so ? don't see anything that would block you from doing it.

ResolverProvider is a global factory, when you call `ResolverProvider.setDefaultResolverFactory` it changes resolverFactory` globally.
Now, if you do that twice, second call will override first factory, but test won't be aware of that and fail, next you will have to debug such test to figure out what happend.
Not only that, if classes were already initialized with old resolver factory, you may get half of classes with old factory, half with new, which is going to create lot's of confusion.
To catch this issue early, we need `ResolverFactory` to throw an exception when this happnes.

Reply:
I believe that it's as we discussed now